mage_ai/frontend/tests/pages.spec.ts

```diff
@@ -0,0 +1,33 @@
+import { expect, test } from './base';
+
+test('ensure all main pages load properly', async ({ page }) => {
+  async function navigateToAndWaitTilLoaded(name: string) {
+    await page.locator('div[class*="indexstyle__VerticalNavigationStyleComponent"] > div').hover();
+    await page.getByRole('link', { name })
+      .and(page.getByTestId('navigation_link'))
```

Comment:
@edmondwinston Ended up needing to just add a `data-testid` attribute to certain link/button elements so we can get it easier in the tests. Simply trying to get a link element by its name wasn't reliable, since there was often another element on the page that conflicted.